app/src/main/java/quotes/App.java

```diff
@@ -18,36 +18,42 @@
 public class App {
          Random random = new Random();
 
-    public String getGreeting() {
-        return "Hello World!";
+    public App(Random random) {
     }
 
-    public String app(String manualPath) throws IOException {
-        System.out.println(new App().getGreeting());
+    public static void main(String[] args) throws IOException {
 
         // Read the quotes from the JSON file
-        manualPath = manualPath != null ? manualPath : "./app/src/main/resources/recentquotes.json";
-        Path path = Paths.get(manualPath);
+        Path path = Paths.get("./app/src/main/resources/recentquotes.json");
         String json = Files.readString(path);
         Gson gson = new GsonBuilder().create();
         Quote[] quotes = gson.fromJson(json, Quote[].class);
 
         // Choose a random quote
+        Random random = new Random();
         Quote randomQuote = quotes[random.nextInt(quotes.length)];
 
         // Output the author and text of the random quote
         System.out.println("Author: " + randomQuote.getAuthor());
         System.out.println("Text: " + randomQuote.getText());
-        return randomQuote.getAuthor();
+
+
+    }
+
+    public Quote getRandomQuote() throws IOException {
+        // Read the quotes from the JSON file
+        Path path = Paths.get("/Users/dashalewis/projects/courses/code-401-java/quotes2/quotes2/app/src/main/resources/recentquotes.json");
```

Comment:
Java can find a lot of this file path info for you. In this case, this line could be rewritten as

`Path path = Paths.get(System.getProperty("user.dir") + "/src/main/resources/recentquotes.json");`

This will find the User directory on whatever computer the app is being run on, and then add the specific path to the recentquotes file.

*NOTE:* This only works if you run the app from the command line. If you run the app from IntelliJ, the path should be

`Path path = Paths.get(System.getProperty("user.dir") + "/app/src/main/resources/recentquotes.json");`

I don't know why it works that way.